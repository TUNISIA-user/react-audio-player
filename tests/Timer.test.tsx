import React from "react";
import { expect, test } from "vitest";
import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer"
import "@testing-library/jest-dom"
import Timer from "../src/components/Timer";


describe("Test timer", () => {
  test("Timer renders", () => {
    render(<Timer />)
    expect(screen.getByTestId("timer")).toHaveTextContent("--:--")
  })

  test("Timer renders with seconds", () => {
    render(<Timer seconds={23} />)
    expect(screen.getByTestId("timer")).toHaveTextContent("00:23")
  })

  test("Timer renders with seconds more than 60", () => {
    render(<Timer seconds={62} />)
    expect(screen.getByTestId("timer")).toHaveTextContent("01:02")
  })

  test("Timer renders with seconds more than 600", () => {
    render(<Timer seconds={620} />)
    expect(screen.getByTestId("timer")).toHaveTextContent("10:20")
  })

  test("Timer snapshot", () => {
    const timer = renderer.create(<Timer />)
    const tree = timer.toJSON();
    expect(tree).toMatchSnapshot();
  })

  test("Timer snapshot with time passed", () => {
    const timer = renderer.create(<Timer seconds={671}/>)
    const tree = timer.toJSON();
    expect(tree).toMatchSnapshot();
  })
})
