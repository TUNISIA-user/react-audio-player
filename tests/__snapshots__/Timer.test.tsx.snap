// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Test timer > Timer snapshot 1`] = `
<div
  data-testid="timer"
  style={
    {
      "color": "#000000",
      "fontSize": 14,
    }
  }
>
  --:--
</div>
`;

exports[`Test timer > Timer snapshot with time passed 1`] = `
<div
  data-testid="timer"
  style={
    {
      "color": "#000000",
      "fontSize": 14,
    }
  }
>
  11:11
</div>
`;
